server/src/main/java/org/candlepin/model/ContentCurator.java

```diff
@@ -82,16 +86,73 @@ public Content lookupByUuid(String uuid) {
      */
     @SuppressWarnings("unchecked")
     public CandlepinQuery<Content> getContentByProducts(Collection<Product> products) {
-        if (products != null && !products.isEmpty()) {
+        Collection<String> uuids = new HashSet<String>();
+
+        for (Product product : products) {
+            uuids.add(product.getUuid());
+        }
+
+        return this.getContentByProductUuids(uuids);
+    }
+
+    /**
+     * Fetches a collection of content used by the given products
+     *
+     * @param productUuids
+     *  A collection of UUIDs representing the products for which to fetch content
+     *
+     * @return
+     *  A collection of content used by the specified products
+     */
+    @SuppressWarnings("unchecked")
+    public CandlepinQuery<Content> getContentByProductUuids(Collection<String> productUuids) {
+        if (productUuids != null && !productUuids.isEmpty()) {
             // We're doing this in two queries because (a) that's what Hibernate's doing already due
             // to the projection and (b) DISTINCT_ROOT_ENTITY only works when listing, not when
             // scrolling.
             Session session = this.currentSession();
 
-            List<String> uuids = session.createCriteria(ProductContent.class)
-                .add(CPRestrictions.in("product", products))
-                .setProjection(Projections.distinct(Projections.property("content.uuid")))
-                .list();
+            // For reasons I can only speculate, Hibernate literally refuses* to run this query
+            // when built with HQL or Criteria, so we're doing it in straight SQL.
+            // * By refuses, I mean silently returns an empty list without ever actually hitting
+            // the database.
+            Collection<String> uuids = new HashSet<String>();
+
+            // Make sure we don't hit the parameter limit when building queries...
+            Iterable<List<String>> uuidBlocks = Iterables.partition(productUuids,
+                AbstractHibernateCurator.QUERY_PARAMETER_LIMIT);
+
+            for (List<String> uuidBlock : uuidBlocks) {
+                StringBuilder builder = new StringBuilder("SELECT DISTINCT content_uuid FROM ")
+                    .append(ProductContent.DB_TABLE)
+                    .append(" WHERE (");
+
+                int blockCount = (int) Math.ceil(uuidBlock.size() /
+                    (float) AbstractHibernateCurator.IN_OPERATOR_BLOCK_SIZE);
+
+                for (int i = 0; i < blockCount;) {
+                    if (i != 0) {
+                        builder.append(" OR ");
+                    }
+
+                    builder.append("product_uuid IN (?").append(++i).append(')');
+                }
+
+                builder.append(')');
+
+                SQLQuery query = session.createSQLQuery(builder.toString());
```

Comment:
Due to complex nature of building up this query, lets add a quick debug statement showing the resulting query.